docs/news/posts/2025-09-05-orm.md

```diff
@@ -0,0 +1,474 @@
+---
+blogpost: true
+category: Blog
+tags: architecture
+author: Julian Geiger
+date: 2025-09-05
+---
+
+# Understanding AiiDA's ORM architecture
+
+In this post, we'll take a deep dive into the implementation of AiiDA's Object-Relational Mapping (ORM) system.
+We'll explore how AiiDA leverages [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy) to create a flexible (ORM) backend that abstracts database operations and separates concerns between the user-facing Python objects and the underlying database persistence.
+
+## Architecture overview: The multi-layer design
+
+AiiDA's ORM follows a four-layer architecture that provides clean separation between the user interface, business logic, and data persistence:
+
+```
+┌─────────────────────┐
+│   User interface    │  ← Node (Python ORM class)
+│     (orm/nodes)     │
+├─────────────────────┤
+│  Backend interface  │  ← BackendNode (Abstract base class)
```

Comment:
I don't full understand it but backend interface is not a good name since you also sometimes refer to the database as backend. Is this like a `orm node backend`?